### PR TITLE
Fix automatic event participation when joining/quitting a serie

### PR DESCRIPTION
**Pull request by AlexisPDS**
## Summary

This PR implements automatic event participation management when users join or quit a serie. When a user joins a
  serie, they are now automatically added to all events in that serie. Similarly, when quitting a serie, the user is
   removed from all associated events.

##  Changes

  SerieDetailsViewModel.kt

  - joinSerie: Now fetches all events in the serie and adds the user to each event's participant list before
  updating the serie
  - quitSerie: Now fetches all events in the serie and removes the user from each event's participant list before
  updating the serie

  SerieDetailsViewModelTest.kt

  Updated existing tests to verify event participation updates:
  - joinSerie_validUser_joinsSuccessfully: Now verifies user is added to all events
  - quitSerie_regularParticipant_quitsSuccessfully: Now verifies user is removed from all events
  - joinAndQuitSerie_workflow_worksCorrectly: Now verifies complete join/quit workflow affects events

  Added new comprehensive test coverage:
  - joinSerie_addsUserToAllEventsInSerie: Tests joining with 3 events, verifies participant counts
  - joinSerie_withNoEvents_stillJoinsSerieSuccessfully: Edge case for series with no events
  - quitSerie_removesUserFromAllEventsInSerie: Tests quitting with 3 events, verifies other participants remain
  - quitSerie_withNoEvents_stillQuitsSerieSuccessfully: Edge case for series with no events

## Notes

Fix issue #268

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt [hunk 1]
```diff
@@ -175,6 +175,12 @@ class SerieDetailsViewModel(
       // Add user to participants
       if (_uiState.value.canJoin(currentUserId)) {
         val updatedParticipants = serie.participants + currentUserId
+        val events = eventsRepository.getEventsByIds(serie.eventIds)
+        events.forEach { event ->
+          val updatedEventParticipants = event.participants + currentUserId
```
Comment (MathieuPFR) [inline app/src/main/java/com/android/joinme/ui/overview/SerieDetailsViewModel.kt]: Current join code does: val updatedEventParticipants = event.participants + currentUserId
That appends the id even if it’s already present, producing duplicates. While your quit logic removes all occurrences, duplicates may still cause wrong participant counts or UI duplication until quit runs.
Suggestion: ensure uniqueness when adding. Example:
val updatedEventParticipants = (event.participants + currentUserId).distinct()
or check contains before adding:
if (!event.participants.contains(currentUserId)) { ... }
